.github/workflows/requirements.yml

```diff
@@ -0,0 +1,66 @@
+# cspell:ignore noreply
+
+name: Upgrade requirements
+
+on:
+  schedule:
+    - cron: "0 2 * * 1"
+
+jobs:
+  requirements:
+    name: Upgrade requirement files
+    continue-on-error: true
+    runs-on: ubuntu-18.04
+    strategy:
+      matrix:
+        python-version:
+          - 3.6
+          - 3.7
+          - 3.8
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@master
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pip-tools
+      - name: Upgrade dependencies
+        run: bash dev/upgrade.sh
```

Comment:
So this will return an error code when there are no changes, right? So any actual error in `upgrade.sh` will be "skipped".

It would be better if we could decide whether to make a new PR based on a diff of the requirement files.

Reply:
No, it will always upgrade. But if there are no file changes (in this cron job), there is nothing to commit, and the push job afterwards won't create a PR.

Reply:
So then why is there a `continue-on-error` in the workflow?

Reply:
There is no error (I assume you refer to the `exit` codes in the `upgrade.sh` script). Unless it's something fatal like `pip-compile` not existing or being unable to determine the Python version.

Reply:
Ah ok `continue-on-error` is there so that the 3.6, 3.7 and 3.8 are all run if there is some problem. I did that for debugging purposes when there were still conflicts in the `txt` files.

Reply:
The problem I have now is that the workflow setting `continue-on-error` will just skip this build if there are errors and not really pop up as red, probably as gray (maybe green). So I don't understand why `continue-on-error` is there/

Reply:
GitHub kills the other jobs in the matrix is one of them fails. But you're right, the push job would still run if there are some problems that still result in are changes to commit. 65198e3 should take care of it, but perhaps it's indeed better to remove this `continue-on-error`.

Reply:
Done 925f344 :heavy_check_mark: